zebra-state/src/service/read/find.rs

```diff
@@ -107,7 +107,7 @@ pub fn non_finalized_state_contains_block_hash(
     non_finalized_state: &NonFinalizedState,
     hash: block::Hash,
 ) -> Option<KnownBlock> {
-    let mut chains_iter = non_finalized_state.chain_set.iter().rev();
+    let mut chains_iter = non_finalized_state.chain_iter();
```

Comment:
This needs to be reversed here to correctly return `BestChain` or `SideChain`, I think it's optional everywhere else it's been updated (all tests) so we could also return `self.chain_set.iter().rev()` from `chain_iter()`.

```suggestion
    let mut chains_iter = non_finalized_state.chain_iter().rev();
```

Reply:
Thanks, that was the original implementation, but I didn't catch it in the cherry-pick.

Fixed in de1269f59f6edb1a032b39a4ba40b70c87424b6c.